tests/post/steps/test_volume.py

```diff
@@ -97,10 +104,88 @@ def test_volume_delete_while_pending(host, teardown):
     pass
 
 
+@scenario(
+    "../features/volume.feature", "Test volume creation (sparseLoopDevice Block mode)"
+)
+def test_volume_creation_loop_block(host, teardown):
+    pass
+
+
+@scenario(
+    "../features/volume.feature", "Test volume creation (rawBlockDevice Block mode)"
+)
+def test_volume_creation_rawblock_block(host, teardown):
+    pass
+
+
+@scenario(
+    "../features/volume.feature", "Test volume creation (lvmLogicalVolume Block mode)"
+)
+def test_volume_creation_lvmlv_block(host, teardown):
+    pass
+
+
 # }}}
 # Given {{{
 
 
+@given("a device exist")
+def device_exist(context, host):
+    context["device_size"] = "20Gi"
+
+    with host.sudo():
+        cmd_ret = host.check_output("salt-call --out json --local temp.file")
+
+    sparse_file = json.loads(cmd_ret)["local"]
+
+    with host.sudo():
+        host.check_output(
+            "salt-call --local file.truncate {} {}".format(
+                sparse_file, _quantity_to_bytes(context["device_size"])
+            )
+        )
+        cmd_ret = host.check_output(
+            "salt-call --out=json --local cmd.run 'losetup -fP --show {}'".format(
+                sparse_file
+            )
+        )
+
+    context["device_path"] = json.loads(cmd_ret)["local"]
+    context["device_name"] = os.path.basename(context["device_path"])
+
+    yield
+
+    with host.sudo():
+        host.check_output(
+            "salt-call --local cmd.run 'losetup -d {}'".format(context["device_path"])
+        )
+        host.check_output("salt-call --local file.remove {}".format(sparse_file))
+
+
+@given(parsers.parse("a LVM VG '{name}' exist"))
+def lvm_vg_exist(context, device_exist, host, name):
+    assert context["device_path"]
```

Comment:
May be worth to add an error message, to help if someone forgets to add `Given a device exists`.

Reply:
You do not need this `Given` btw we use `device_exists` as a fixture here

Reply:
Uh, and it works out of the box if this fixture is decorated as a `@given` as well? OK, fine by me :+1: just wondering if we care about this assertion then.

Reply:
Yes it works every given, then, ... are fixture :smiley: 
Yes we do not really care I can remove it